client/src/components/core/ms-modal/MsModal.vue

```diff
@@ -74,8 +82,17 @@
 </template>
 
 <script lang="ts">
+export enum MsModalTheme {
+  Info = 'modal-info',
+  Success = 'modal-success',
+  Warning = 'modal-warning',
+  Error = 'modal-error',
+}
+
 export interface MsModalConfig {
-  title: string,
+  title?: string,
+  titleIcon?: string,
+  theme?: MsModalTheme,
```

Comment:
I feel like `theme` and `titleIcon` are too similar and clash with one another a bit (like you can have a warning icon with a success theme). I would be in favor of dropping the icon, making the theme mandatory, and letting the component choose what icon it wants to display depending on the theme. It gives us a more reliable API, which means less options for devs like me to screw up the UI. Of course it also means that you could not customise the modal, but I think it's a good thing to be more restrictive.

Reply:
Done, some code could be common between the 2 files, but the same applies to the CSS.
To keep focus of the issue, I'll keep it simple and make 2 files with their code each.

Maybe it's even better to keep them totally distincts (MsModal and MsAlertModal)